tests/purs/failing/InfiniteKind2.out

```diff
@@ -10,7 +10,7 @@ at tests/purs/failing/InfiniteKind2.purs:5:23 - 5:27 (line 5, column 23 - line 5
 while checking that type [33mTree[0m
   has kind [33mt0[0m
 while inferring the kind of [33mm Tree[0m
-in type constructor [33mTree[0m
+in data binding group Tree
```

Comment:
I'm not sure what this change is about.

Reply:
Oh I see. I took out the singleton synonym check. I think this is actually correct now since it is a cyclical type.